drivers/sx127x/sx127x.c

```diff
@@ -0,0 +1,491 @@
+    netdev_t *netdev = (netdev_t*) dev;
+
+    bool freq_hop_on = SX127X_FREQUENCY_HOPPING;
+    bool iq_invert = SX127X_IQ_INVERSION;
+    uint8_t rx_single = SX127X_RX_SINGLE;
+    uint32_t tx_timeout = SX127X_TX_DEFAULT_TIMEOUT;
+
+    netdev->driver->set(netdev, NETOPT_LORA_HOP,
+                        &freq_hop_on, sizeof(bool));
+    netdev->driver->set(netdev, NETOPT_LORA_IQ_INVERT,
+                        &iq_invert, sizeof(bool));
+    netdev->driver->set(netdev, NETOPT_LORA_SINGLE_RECEIVE,
+                        &rx_single, sizeof(uint8_t));
+    netdev->driver->set(netdev, NETOPT_LORA_TX_TIMEOUT,
+                        &tx_timeout, sizeof(uint32_t));
+
+    lora_settings.bandwidth = SX127X_BW_DEFAULT;
+    lora_settings.cr = SX127X_CR_DEFAULT;
+    lora_settings.sf = SX127X_SF_DEFAULT;
+    lora_settings.crc_on = SX127X_PAYLOAD_CRC_ON;
+    lora_settings.freq_hop_period = SX127X_FREQUENCY_HOPPING_PERIOD;
+    lora_settings.implicit_header_mode = SX127X_IMPLICIT_HEADER_MODE;
+    lora_settings.payload_len = SX127X_PAYLOAD_LENGTH;
+    lora_settings.power = SX127X_RADIO_POWER;
+    lora_settings.preamble_len = SX127X_PREAMBLE_LENGTH;
+    lora_settings.rx_timeout = SX127X_SYMBOL_TIMEOUT;
+
+    /* Copy LoRa configuration into device structure */
+    memcpy(&dev->settings.lora, &lora_settings, sizeof(sx127x_lora_settings_t));
+
+    sx127x_set_modem(dev, SX127X_MODEM_LORA);
+    sx127x_set_bandwidth(dev, dev->settings.lora.bandwidth);
+    sx127x_set_coding_rate(dev, dev->settings.lora.cr);
+    sx127x_set_implicit_header_mode(dev, dev->settings.lora.implicit_header_mode);
+    sx127x_set_spreading_factor(dev, dev->settings.lora.sf);
+    sx127x_set_crc(dev, dev->settings.lora.crc_on);
+    sx127x_set_symbol_timeout(dev, dev->settings.lora.rx_timeout);
+    sx127x_set_preamble_length(dev, dev->settings.lora.preamble_len);
+    sx127x_set_payload_length(dev, dev->settings.lora.payload_len);
+    sx127x_set_hop_period(dev, dev->settings.lora.freq_hop_period);
+    sx127x_set_power(dev, dev->settings.lora.power);
+
+    sx127x_set_channel(dev, LORA_ISM_FREQUENCY);
+}
+
+uint32_t sx127x_random(sx127x_t *dev)
+{
+    uint32_t rnd = 0;
+
+    sx127x_set_modem(dev, SX127X_MODEM_LORA); /* Set LoRa modem ON */
+
+    /* Disable LoRa modem interrupts */
+    sx127x_reg_write(dev, SX127X_REG_LR_IRQFLAGSMASK, SX127X_RF_LORA_IRQFLAGS_RXTIMEOUT |
+                     SX127X_RF_LORA_IRQFLAGS_RXDONE |
+                     SX127X_RF_LORA_IRQFLAGS_PAYLOADCRCERROR |
+                     SX127X_RF_LORA_IRQFLAGS_VALIDHEADER |
+                     SX127X_RF_LORA_IRQFLAGS_TXDONE |
+                     SX127X_RF_LORA_IRQFLAGS_CADDONE |
+                     SX127X_RF_LORA_IRQFLAGS_FHSSCHANGEDCHANNEL |
+                     SX127X_RF_LORA_IRQFLAGS_CADDETECTED);
+
+    sx127x_set_op_mode(dev, SX127X_RF_OPMODE_STANDBY);
+
+    /* Set radio in continuous reception */
+    sx127x_set_op_mode(dev, SX127X_RF_OPMODE_RECEIVER);
```

Comment:
apparently not (comparing to [LoRaMac-node/sx1276.c](https://github.com/Lora-net/LoRaMac-node/blob/master/src/radio/sx1276/sx1276.c#L306)